test.lua

```diff
@@ -92,6 +92,11 @@ local function main()
         local rootUri = furi.encode(TESTROOT)
         client:initialize {
             rootUri = rootUri,
+            capabilities = {
+               workspace = {
+                  configuration = true,
+               }
+            },
```

Comment:
The changes in this PR seems to be very simple (at least not too much logic changes), I am just surprised why it breaks all those tests 🤔 

- The main change in the tests is just adding this new option `workspace.configuration` when initializing the fake client:
```lua
    client:initialize {
        capabilities = {
            workspace = {
                configuration = true,
            }
        }
    }
```
- Then I looked into this `client:initialize` implementation, it will merge the `params` with the `defaultClientOptions` which defines some other capabilities:
https://github.com/LuaLS/lua-language-server/blob/a126e9c4850726f86d8480aacbb8ec23c1b158ef/script/lclient.lua#L66-L94
- It is using the `util.tableMerge`:
https://github.com/LuaLS/lua-language-server/blob/a126e9c4850726f86d8480aacbb8ec23c1b158ef/script/utility.lua#L950-L955
- ‼️ we can see that it's just doing a merge with the **1st layer**, and due to `defaultClientOptions` already has the key `capabilities`, your newly added `capabilities = { workspace = { configuration = true } }` will be overridden => no effect 🙈

---

I think you have to add the `workspace.configuration = true` in the `defaultClientOptions` in [script/lclient.lua](https://github.com/LuaLS/lua-language-server/blob/master/script/lclient.lua#L66-L87) to make it work. Then you don't have to add it in every test (which actually won't work due to the above `util.tableMerge` logic)
```lua
local defaultClientOptions = {
    ...
    capabilities = {
        ...
        },
        workspace = {
            configuration = true,
        },
    },
}
```
- i.e. after changing this `defaultClientOptions`, you can **revert** all changes in `test*` files (I suggest to use amend & force push, this can avoid leaving useless history in the git tree)

I tested this and it can pass all tests now (including the `full` test) 😄 @nospam2998

Reply:
I'm no wiser on what the tests actually are designed to check for, but your suggested change sure made them pass. Guess that's good enough for me. Thanks @tomlau10!